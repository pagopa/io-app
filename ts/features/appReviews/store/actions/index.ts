import { ActionType, createStandardAction } from "typesafe-actions";
import { AppFeedbackUri } from "../../../../../definitions/content/AppFeedbackUri";

export type TopicKeys = keyof AppFeedbackUri;

export const appReviewPositiveFeedback = createStandardAction(
  "APP_REVIEW_POSITIVE_FEEDBACK"
)();

export const appReviewNegativeFeedback = createStandardAction(
  "APP_REVIEW_NEGATIVE_FEEDBACK"
)<TopicKeys>();

// Reset feedback data just for testing purposes
export const clearFeedbackDatas = createStandardAction(
  "APP_REVIEW_RESET_FEEDBACK"
)();

export type AppFeedbackActions =
  | ActionType<typeof appReviewPositiveFeedback>
  | ActionType<typeof appReviewNegativeFeedback>
  | ActionType<typeof clearFeedbackDatas>;
