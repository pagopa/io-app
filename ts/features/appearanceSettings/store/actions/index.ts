/**
 * Action types and action creator related to the Appearance settings.
 */
import { ActionType, createStandardAction } from "typesafe-actions";

export const setShowAppearanceSettingsBanner = createStandardAction(
  "SHOW_APPEARANCE_SETTINGS_BANNER"
)<boolean>();

export type AppearanceSettingsActions = ActionType<
  typeof setShowAppearanceSettingsBanner
>;
