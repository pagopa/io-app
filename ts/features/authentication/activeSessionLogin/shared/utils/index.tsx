import I18n from "i18next";
import { Millisecond } from "@pagopa/ts-commons/lib/units";
import { Platform } from "react-native";
import { CieAuthenticationErrorReason } from "../../../login/cie/store/actions/cie";
import { ReadingState } from "../../../login/cie/components/CieCardReadingAnimation";

// the timeout we sleep until move to consent form screen when authentication goes well
export const WAIT_TIMEOUT_NAVIGATION = 1700 as Millisecond;
export const WAIT_TIMEOUT_NAVIGATION_ACCESSIBILITY = 5000 as Millisecond;
export const VIBRATION = 100 as Millisecond;
export const accessibityTimeout = 100 as Millisecond;

// A subset of Cie Events (errors) which is of interest to analytics
export const analyticActions = new Map<CieAuthenticationErrorReason, string>([
  // Reading interrupted before the sdk complete the reading
  ["Transmission Error", I18n.t("authentication.cie.card.error.onTagLost")],
  ["ON_TAG_LOST", I18n.t("authentication.cie.card.error.onTagLost")],
  [
    "TAG_ERROR_NFC_NOT_SUPPORTED",
    I18n.t("authentication.cie.card.error.unknownCardContent")
  ],
  [
    "ON_TAG_DISCOVERED_NOT_CIE",
    I18n.t("authentication.cie.card.error.unknownCardContent")
  ],
  ["PIN Locked", I18n.t("authentication.cie.card.error.generic")],
  ["ON_CARD_PIN_LOCKED", I18n.t("authentication.cie.card.error.generic")],
  ["ON_PIN_ERROR", I18n.t("authentication.cie.card.error.tryAgain")],
  ["PIN_INPUT_ERROR", ""],
  ["CERTIFICATE_EXPIRED", I18n.t("authentication.cie.card.error.generic")],
  ["CERTIFICATE_REVOKED", I18n.t("authentication.cie.card.error.generic")],
  ["AUTHENTICATION_ERROR", I18n.t("authentication.cie.card.error.generic")],
  [
    "EXTENDED_APDU_NOT_SUPPORTED",
    I18n.t("authentication.cie.nfc.apduNotSupported")
  ],
  [
    "ON_NO_INTERNET_CONNECTION",
    I18n.t("authentication.cie.card.error.tryAgain")
  ],
  ["STOP_NFC_ERROR", ""],
  ["START_NFC_ERROR", ""]
]);

export type TextForState = {
  title: string;
  subtitle?: string;
  content: string;
};

// some texts changes depending on current running Platform
export const getTextForState = (
  state: ReadingState.waiting_card | ReadingState.error,
  errorMessage: string = ""
): TextForState => {
  const texts: Record<
    ReadingState.waiting_card | ReadingState.error,
    TextForState
  > = Platform.select({
    ios: {
      [ReadingState.waiting_card]: {
        title: I18n.t("authentication.cie.card.titleiOS"),
        subtitle: I18n.t("authentication.cie.card.layCardMessageHeaderiOS"),
        // the native alert hides the screen content and shows a message it self
        content: ""
      },
      [ReadingState.error]: {
        title: I18n.t("authentication.cie.card.error.readerCardLostTitle"),
        subtitle: "",
        // the native alert hides the screen content and shows a message it self
        content: ""
      },
      [ReadingState.reading]: {
        title: I18n.t("authentication.cie.card.titleiOS"),
        subtitle: I18n.t("authentication.cie.card.layCardMessageHeaderiOS"),
        // the native alert hides the screen content and shows a message it self
        content: ""
      }
    },
    default: {
      [ReadingState.waiting_card]: {
        title: I18n.t("authentication.cie.card.title"),
        subtitle: I18n.t("authentication.cie.card.layCardMessageHeader"),
        content: I18n.t("authentication.cie.card.layCardMessageFooter")
      },
      [ReadingState.error]: {
        title: I18n.t("authentication.cie.card.error.readerCardLostTitle"),
        subtitle: I18n.t("authentication.cie.card.error.onTagLost"),
        content: errorMessage
      }
    }
  });
  return texts[state];
};
