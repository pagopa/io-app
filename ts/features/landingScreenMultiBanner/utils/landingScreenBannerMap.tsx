import * as React from "react";
import { SettingsDiscoveryBanner } from "../../../screens/profile/components/SettingsDiscoveryBanner";
import { GlobalState } from "../../../store/reducers/types";
import { ItwDiscoveryBanner } from "../../itwallet/common/components/discoveryBanner/ItwDiscoveryBanner";
import { isItwDiscoveryBannerRenderableSelector } from "../../itwallet/common/store/selectors";
import { hasUserAcknowledgedSettingsBannerSelector } from "../../profileSettings/store/selectors";

type ComponentWithCloseHandler = (
  closeHandler: () => void
) => React.ReactElement;
type ComponentAndLogic = {
  component: ComponentWithCloseHandler;
  isRenderableSelector: (state: GlobalState) => boolean;
};

export type BannerMapById = {
  [key in LandingScreenBannerId]: ComponentAndLogic;
};
export type LandingScreenBannerId =
  keyof typeof LANDING_SCREEN_BANNERS_ENABLED_MAP;

export const LANDING_SCREEN_BANNERS_ENABLED_MAP = {
  ITW_DISCOVERY: true,
  SETTINGS_DISCOVERY: true
} as const;

export const landingScreenBannerMap: BannerMapById = {
  ITW_DISCOVERY: {
    component: closeHandler => (
      <ItwDiscoveryBanner closable handleOnClose={closeHandler} />
    ),
    isRenderableSelector: isItwDiscoveryBannerRenderableSelector
  },
  SETTINGS_DISCOVERY: {
    component: closeHandler => (
      <SettingsDiscoveryBanner handleOnClose={closeHandler} />
    ),
    isRenderableSelector: (state: GlobalState) =>
      !hasUserAcknowledgedSettingsBannerSelector(state)
  }
} as const;
