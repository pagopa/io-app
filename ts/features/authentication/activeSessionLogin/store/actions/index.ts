import { ActionType, createStandardAction } from "typesafe-actions";
import { SpidIdp } from "../../../../../utils/idps";
import { SessionToken } from "../../../../../types/SessionToken";

export const setActiveSessionLoginLocalFlag = createStandardAction(
  "SET_ACTIVE_SESSION_LOGIN_LOCAL_FLAG"
)<boolean>();

export const setStartActiveSessionLogin = createStandardAction(
  "SET_START_ACTIVE_SESSION_LOGIN"
)();

export const setRetryActiveSessionLogin = createStandardAction(
  "SET_RETRY_ACTIVE_SESSION_LOGIN"
)();

export const setIdpSelectedActiveSessionLogin = createStandardAction(
  "SET_IDP_SELECTED_ACTIVE_SESSION_LOGIN"
)<SpidIdp>();

export const activeSessionLoginFailure = createStandardAction(
  "ACTIVE_SESSION_LOGIN_FAILURE"
)();

export const activeSessionLoginSuccess = createStandardAction(
  "ACTIVE_SESSION_LOGIN_SUCCESS"
)<SessionToken>();

export const setFastLoginOptSessionLogin = createStandardAction(
  "SET_FAST_LOGIN_OPT_IN_ACTIVE_SESSION_LOGIN"
)<boolean>();

export const consolidateActiveSessionLoginData = createStandardAction(
  "CONSOLIDATE_ACTIVE_SESSION_LOGIN_DATA"
)<{
  token: SessionToken;
  idp: SpidIdp;
  fastLoginOptIn: boolean;
}>();

export const setFinishedActiveSessionLoginFlow = createStandardAction(
  "SET_FINISHED_ACTIVE_SESSION_LOGIN_FLOW"
)();

export const setLoggedOutUserWithDifferentCF = createStandardAction(
  "SET_LOGGED_OUT_USER_WITH_DIFFERENT_CF"
)();

export type LoginInfoActions =
  | ActionType<typeof setActiveSessionLoginLocalFlag>
  | ActionType<typeof setStartActiveSessionLogin>
  | ActionType<typeof setRetryActiveSessionLogin>
  | ActionType<typeof setIdpSelectedActiveSessionLogin>
  | ActionType<typeof activeSessionLoginFailure>
  | ActionType<typeof activeSessionLoginSuccess>
  | ActionType<typeof consolidateActiveSessionLoginData>
  | ActionType<typeof setFastLoginOptSessionLogin>
  | ActionType<typeof setFinishedActiveSessionLoginFlow>
  | ActionType<typeof setLoggedOutUserWithDifferentCF>;
