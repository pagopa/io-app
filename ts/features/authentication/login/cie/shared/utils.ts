import { Platform } from "react-native";
import { isDevEnv } from "../../../../../utils/environment";
import { SpidLevel } from "../utils";

export type WebViewLoginNavigationProps = {
  spidLevel: SpidLevel;
  isUat: boolean;
};

export const iOSUserAgent =
  "Mozilla/5.0 (iPhone; CPU iPhone OS 14_0_1 like Mac OS X) AppleWebKit/605.1.15 (KHTML, like Gecko) Version/14.0 Mobile/15E148 Safari/604.1";

export const defaultUserAgent = Platform.select({
  ios: iOSUserAgent,
  default: undefined
});

export const originSchemasWhiteList = [
  "https://*",
  "iologin://*",
  ...(isDevEnv ? ["http://*"] : [])
];

export const WHITELISTED_DOMAINS = [
  "https://idserver.servizicie.interno.gov.it",
  "https://oidc.idserver.servizicie.interno.gov.it",
  "https://mtls.oidc.idserver.servizicie.interno.gov.it",
  "https://mtls.idserver.servizicie.interno.gov.it",
  "https://ios.idserver.servizicie.interno.gov.it",
  "https://ios.oidc.idserver.servizicie.interno.gov.it"
];

export type CieIdLoginProps = {
  spidLevel: SpidLevel;
  isUat: boolean;
};
