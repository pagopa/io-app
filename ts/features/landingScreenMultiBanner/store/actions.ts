import { ActionType, createStandardAction } from "typesafe-actions";
import { LandingScreenBannerId } from "../utils/landingScreenBannerMap";

type UpdateBannerVisibilityPayloadType = {
  id: LandingScreenBannerId;
  enabled: boolean;
};
export const updateLandingScreenBannerVisibility = createStandardAction(
  "UPDATE_LANDING_SCREEN_BANNER_STATE"
)<UpdateBannerVisibilityPayloadType>();

export type LandingScreenBannerActions = ActionType<
  typeof updateLandingScreenBannerVisibility
>;
