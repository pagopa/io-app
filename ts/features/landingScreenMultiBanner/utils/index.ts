import { mapValues } from "lodash";
import { BannerMapById } from "./landingScreenBannerMap";

/**
 * extracts renderability selectors from component map,
 * and indexes them by ID
 * ----
 * @param bannerMap
 * a `{[bannerID]:{ isRenderableSelector: Selector, foo:bar } }` map
 * @returns
 * a `{[bannerID]: isRenderableSelector }` map
 */
export const renderabilitySelectorsFromBannerMap = (bannerMap: BannerMapById) =>
  mapValues(bannerMap, ({ isRenderableSelector }) => isRenderableSelector);
