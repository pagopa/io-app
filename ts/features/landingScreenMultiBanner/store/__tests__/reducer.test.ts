import { applicationChangeState } from "../../../../store/actions/application";
import { LandingScreenBannerId } from "../../utils/landingScreenBannerMap";
import { updateLandingScreenBannerVisibility } from "../actions";
import {
  landingScreenBannersReducer,
  LandingScreenBannerState
} from "../reducer";

describe("landingScreenBannersReducer", () => {
  it("should match snapshot", () => {
    expect(
      landingScreenBannersReducer(undefined, applicationChangeState("active"))
    ).toMatchSnapshot("undefined_no_action");
  });
  it("should update state on action received -- enabled:false", () => {
    const baseState = {
      "banner-1": true
    } as unknown as LandingScreenBannerState;
    const action = updateLandingScreenBannerVisibility({
      enabled: false,
      id: "banner-1" as LandingScreenBannerId
    });

    const reducer = landingScreenBannersReducer(baseState, action);
    expect(reducer).toEqual({
      "banner-1": false
    });
  });
  it("should update state on action received -- enabled:true", () => {
    const baseState = {
      "banner-1": false
    } as unknown as LandingScreenBannerState;
    const action = updateLandingScreenBannerVisibility({
      enabled: true,
      id: "banner-1" as LandingScreenBannerId
    });

    const reducer = landingScreenBannersReducer(baseState, action);
    expect(reducer).toEqual({
      "banner-1": true
    });
  });
  it("should not update state on invalid action received -- invalid id", () => {
    const baseState = {
      "banner-1": false
    } as unknown as LandingScreenBannerState;
    const action = updateLandingScreenBannerVisibility({
      enabled: true,
      id: "banner-5" as LandingScreenBannerId
    });

    const reducer = landingScreenBannersReducer(baseState, action);
    expect(reducer).toEqual(baseState);
  });
});
