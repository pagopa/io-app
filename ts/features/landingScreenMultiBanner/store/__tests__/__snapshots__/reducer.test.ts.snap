// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`landingScreenBannersReducer should match snapshot: undefined_no_action 1`] = `
{
  "ITW_DISCOVERY": true,
  "SETTINGS_DISCOVERY": true,
}
`;
